### PR TITLE
ROKKA-169 Search Stacks

### PR DESCRIPTION
**Pull request by pedro-couto**
- [x] add search stack styling
- [x] autocomplete and filter results

### CODE REVIEW

File: src/components/Sidebar.js [hunk 1]
```diff
@@ -15,16 +24,38 @@ class Sidebar extends Component {
     }
   }
 
+  componentWillReceiveProps (nextProps) {
+    this.setState({
```
Comment (mweibel) [inline src/components/Sidebar.js]: add an if statement above to make sure `nextProps.stacks.items` exists

File: src/components/Sidebar.js [hunk 2]
```diff
@@ -15,16 +24,38 @@ class Sidebar extends Component {
...
+      items: nextProps.stacks.items
+    })
+  }
+
+  filterStacks (e) {
+    const query = e.currentTarget.value
```
Comment (mweibel) [inline src/components/Sidebar.js]: does it need currentTarget or would target suffice?

File: src/components/Sidebar.js [hunk 3]
```diff
@@ -15,16 +24,38 @@ class Sidebar extends Component {
...
+    const $items = this.props.stacks.items.filter((item) => {
```
Comment (mweibel) [inline src/components/Sidebar.js]: `$items` are used only when the array contains react elements, which it doesn't in this case. Remove the `$` please

File: src/components/Sidebar.js [hunk 4]
```diff
@@ -15,16 +24,38 @@ class Sidebar extends Component {
...
+      if (item.name.indexOf(query) > -1) {
+        return item.name
+      }
```
Comment (mweibel) [inline src/components/Sidebar.js]: `.filter` just should return true or false. So you can simplify this using `return item.name.indexOf(query) > -1`

File: src/components/Sidebar.js [hunk 5]
```diff
@@ -15,16 +24,38 @@ class Sidebar extends Component {
...
     const showStacks = routePath.indexOf('/stack') >= 0 || routePath === '/new-stack'
-    const { items = [], currentOffset = 0, total = 0 } = this.props.stacks
+    const { currentOffset = 0, total = 0 } = this.props.stacks
 
-    const $stacks = items.map((stack) => {
-      return (
-        <Link key={stack.name} to={`/stacks/${stack.name}`} activeClassName="is-active" className="rka-sidebar-sublink txt-ellipsis">{stack.name}</Link>
-      )
-    })
+    let $stacks
```
Comment (mweibel) [inline src/components/Sidebar.js]: ` = null` missing

File: src/components/Sidebar.js [hunk 6]
```diff
@@ -15,16 +24,38 @@ class Sidebar extends Component {
...
+    if (this.state.items) {
```
Comment (mweibel) [inline src/components/Sidebar.js]: do you need the `if`? as the constructor also sets a default for `this.state.items` I assume the if is not needed

File: src/components/forms/FormGroup.js [hunk 7]
```diff
@@ -2,8 +2,8 @@ import React, { PropTypes } from 'react'
 import cx from 'classnames'
 import RequiredIndicator from './RequiredIndicator'
 
-const FormGroup = ({ label, children, required = false, error = null }) => (
-  <div className={cx('rka-form-group', {'has-error': !!error})}>
+const FormGroup = ({ label, children, classname = null, required = false, error = null }) => (
+  <div className={cx('rka-form-group', classname, {'has-error': !!error})}>
```
Comment (mweibel) [inline src/components/forms/FormGroup.js]: `className` is how React does it and so should we

File: src/components/Sidebar.js (hunk 6 above)
Comment (pedro-couto) [inline src/components/Sidebar.js]: the map will complain and give a error.

Comment (pedro-couto) [inline src/components/Sidebar.js]: fixed

File: src/components/Sidebar.js [hunk 8]
```diff
@@ -15,12 +24,31 @@ class Sidebar extends Component {
...
+  }
+
+  filterStacks (e) {
+    const query = e.target.value
+    const items = this.props.stacks.items.filter((item) => {
+      return item.name.indexOf(query) > -1
+    })
+
+    this.setState({
+      items: items
```
Comment (mweibel) [inline src/components/Sidebar.js]: can just say `items` instead of `items: items`

**Pull request merged**